src/client/list.rs

```diff
@@ -42,86 +42,44 @@ pub(crate) trait ListClientExt {
     fn list_paginated(
         &self,
         prefix: Option<&Path>,
-        delimiter: bool,
-        offset: Option<&Path>,
+        opts: ListOpts,
     ) -> BoxStream<'static, Result<ListResult>>;
-
-    fn list(&self, prefix: Option<&Path>) -> BoxStream<'static, Result<ObjectMeta>>;
-
-    #[allow(unused)]
-    fn list_with_offset(
-        &self,
-        prefix: Option<&Path>,
-        offset: &Path,
-    ) -> BoxStream<'static, Result<ObjectMeta>>;
-
-    async fn list_with_delimiter(&self, prefix: Option<&Path>) -> Result<ListResult>;
 }
 
 #[async_trait]
 impl<T: ListClient + Clone> ListClientExt for T {
     fn list_paginated(
         &self,
         prefix: Option<&Path>,
-        delimiter: bool,
-        offset: Option<&Path>,
+        opts: ListOpts,
     ) -> BoxStream<'static, Result<ListResult>> {
+        let ListOpts {
+            delimiter,
+            offset,
+            extensions,
+        } = opts;
+
         let offset = offset.map(|x| x.to_string());
         let prefix = prefix
             .filter(|x| !x.as_ref().is_empty())
             .map(|p| format!("{}{}", p.as_ref(), crate::path::DELIMITER));
 
         stream_paginated(
             self.clone(),
-            (prefix, offset),
-            move |client, (prefix, offset), token| async move {
+            (prefix, offset, extensions),
+            move |client, (prefix, offset, extensions), token| async move {
                 let (r, next_token) = client
                     .list_request(
                         prefix.as_deref(),
                         delimiter,
                         token.as_deref(),
                         offset.as_deref(),
+                        extensions.clone(),
                     )
                     .await?;
-                Ok((r, (prefix, offset), next_token))
+                Ok((r, (prefix, offset, extensions), next_token))
             },
         )
         .boxed()
     }
-
-    fn list(&self, prefix: Option<&Path>) -> BoxStream<'static, Result<ObjectMeta>> {
```

Comment:
Can we revert this change please, the purpose of ListClientExt is that the ListClient implementations just implement list_request and all the ObjectStore list methods are provided for them.

Once list_opts is required we can remove ListClientExt, with the contents of these methods moving into the ObjectStore default impls, but until then...

Reply:
sure